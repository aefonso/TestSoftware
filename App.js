import {Text, View, TextInput} from 'react-native';
import { Button } from 'react-native';
import { useState } from 'react';

export default () => {
  const [numero, setNumero] = useState(0);
  const [dobro, setDobro] = useState(0);

  function calcularDobro(){
    setDobro(parseInt(numero)*2)
  }

  return (
  <View>
    <Text>Cálculo do Dobro</Text>

    <TextInput placeholder='Digite um número' onChangeText={setNumero}/>

    <Button title="Calcular dobro" onPress={calcularDobro}/>

    <Text>{dobro}</Text>
  </View>
  )
}