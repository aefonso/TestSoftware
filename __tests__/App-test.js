import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native'
import App from '../App';

describe('Teste de Snapshot', () => {
    it('Snapshot de App', () => {
        const tela = renderer.create(<App/>).toJSON();
        expect(tela).toMatchSnapshot();
    })
})

describe('Teste de Componentes', () => {
    it('Teste de Componentes de App', () => {
        const { getByText, getByTestId } = render(<App />);

       // const textoInicial = getByText("CÃ¡lculo do Dobro");
       const textoInicial = getByTestId("entrada")
        expect(textoInicial).toBeDefined();
        // Solucao 2//expect(textoInicial).toBeThruthy();
    })
})