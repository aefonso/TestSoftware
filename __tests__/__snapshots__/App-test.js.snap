// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de Snapshot Snapshot de App 1`] = `
<View>
  <Text>
    Cálculo do Dobro
  </Text>
  <TextInput
    onChangeText={[Function]}
    placeholder="Digite um número"
    testID="entrada"
  />
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Calcular dobro
      </Text>
    </View>
  </View>
  <Text>
    0
  </Text>
</View>
`;
